packages/client/hmi-client/src/services/search.ts

```diff
@@ -0,0 +1,16 @@
+import API from '@/api/api';
+import { AssetType, TerariumAssetThatSupportsAdditionalProperties } from '@/types/Types';
+import { logger } from '@/utils/logger';
+
+export async function searchByAssetType(
+	text: string,
+	assetType: AssetType
+): Promise<TerariumAssetThatSupportsAdditionalProperties[]> {
```

Comment:
This should return the same type than used for `AssetType`.

Reply:
I tried some ways of doing that using generics but it doesn't really work. AssetType is an enum.